src/app/admin/semester/create/page.tsx

```diff
@@ -0,0 +1,153 @@
+"use client";
+import * as React from "react";
+import { useForm, SubmitHandler } from "react-hook-form";
+import { Button } from "@/components/ui/button";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+import { format } from "date-fns";
+import { CalendarIcon } from "lucide-react";
+import { cn } from "@/lib/utils";
+import { Calendar } from "@/components/ui/calendar";
+import { Popover, PopoverContent, PopoverTrigger } from "@/components/ui/popover";
+import apiService from "@/services/api-service";
+import handleFormValidationErrors from "@/lib/handle-form-validation-errors";
+import { toast } from "sonner";
+import { useRouter } from "next/navigation";
+
+interface FormInput {
+	title: string;
+}
+
+export default function TeacherRegistrationForm() {
+	const router = useRouter();
+	const [startDate, setStartDate] = React.useState<Date | undefined>();
+	const [endDate, setEndDate] = React.useState<Date | undefined>();
+
+	const {
+		register,
+		handleSubmit,
+		setError,
+		formState: { errors, isSubmitting },
+	} = useForm<FormInput>();
+
+	const onSubmit: SubmitHandler<FormInput> = async (data) => {
+		const token = localStorage.getItem("token");
+		if (!token) {
+			toast.error("You are not authenticated. Please log in first.");
+			router.push("/login");
+			return;
+		}
```

Comment:
We don't need this condition cause we can't access this page if we are not logged in